### ISSUE TITLE
Large File Size Warning Not Displaying On Files Which Are Too Large

### ISSUE THREAD

**Issue by giodestone** [desktop/desktop#11613]
### Describe the bug
When committing a file very close to the 100MB file limit on Windows, the push fails with:
`error: RPC failed; curl 92 HTTP/2 stream 0 was not closed cleanly: CANCEL (err 8)`
`fatal: the remote end hung up unexpectedly`
...which is the same error message if you commit something over 100MB with the Git terminal.

This appears to indicate that the file size calculation for the large file size warning dialog is incorrect, due to not including the extra overhead that git adds to a binary file (if it does) and/or or the way Git (or GitHub) handles files very close to the maximum size.

[File in question.](https://github.com/giodestone/Emotion-Recognition/blob/main/Source/shape_predictor_68_face_landmarks.zip) Must be extracted first.
### Version & OS

2.6.3 

Windows 10 1909 64bit

### Steps to reproduce the behavior

1. Create empty repository on disk.
2. Create empty repository on GitHub
3. [Download this file.](https://github.com/giodestone/Emotion-Recognition/blob/main/Source/shape_predictor_68_face_landmarks.zip)
4. Extract into the empty repository on disk.
5. Create commit with downloaded file.
6. Push.

### Expected behavior

Warning that the committed file is too big to be added to GitHub is displayed.

### Actual behavior

No warning is given, but rather a pop up with a same error to that when you commit a file that is too large using git terminal.

### Screenshots

N/A

### Logs

`
error: RPC failed; curl 55 Send failure: Connection was aborted
fatal: the remote end hung up unexpectedly
Writing objects: 100% (369/369), 240.19 MiB | 420.00 KiB/s
Writing objects: 100% (369/369), 309.16 MiB | 1.68 MiB/s, done.
Total 369 (delta 69), reused 0 (delta 0), pack-reused 0
fatal: the remote end hung up unexpectedly
Everything up-to-date
`

### Additional context

I investigated the file size on an ext4 ElementaryOS VM, the reported file size is larger (99.7MB vs 95.0MB on Windows), however the number of bytes stays the same (99,696,397 bytes).

The repository was initialized using the terminal (`git init`) then added to a blank repository.

**Comment by steveward** [comment]
Thanks for the report @giodestone. I was unable to reproduce this issue using the steps provided. I downloaded the file you shared, committed it, and was successfully able to push it to a test repository on GitHub. 

Normally the large file error that is shown in the logs is this:

`remote: error: File <filename> is 131.36 MB; this exceeds GitHub's file size limit of 100.00 MB`

The error you are encountering is more commonly seen when connection timeouts occur during Git operations. Have you tested this multiple times to see if you are able to consistently reproduce this issue?

**Comment by giodestone** [comment]
Yes, I have tried it multiple times before submitting the report, although only on my computer (and connection).

I have tried it again just now and the upload succeeded. My upload speed is 20Mb/s when testing this time. It appears that it was something other than Git/GitHub Desktop giving me problems when I submitted the report (though I can't confirm what it was - most likely my internet connection). 

Looking back at GitHub Desktop's logs, my upload speed was similar, though decreased to equivalent of 3.84Mb/s at some percentage points (although this is the log for a bigger commit of 309MB with the file in question). I don't appear to have the logs saved from when I was uploading just the file to a test repository but I don't recall that the speed was much different (the 'writing objects' progress got to 99% with reasonable speed readouts, and then stopped for a moment, gave the `RPC Failed`/`remote end hung up unexpectedly` error and proceeded to 100%). I incorrectly assumed this was just some cryptic error relating to the file size as it was very close to the 100MB limit.

I think this issue can be closed, because it appears to be down to something other than GitHub Desktop. Thank you for your time in trying to resolve this issue.
